File: go/authentication/cmd/generate_token/generate_token.go
// Go SDK examples for generating Aurora DSQL authentication tokens

package generate_token

// --8<-- [start:go-generate-token]
import (
	"context"
	"time"

	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/feature/dsql/auth"
)

func GenerateDbConnectAdminAuthToken(ctx context.Context, clusterEndpoint, region string, expiry time.Duration) (string, error) {
	cfg, err := config.LoadDefaultConfig(ctx)
	if err != nil {
		return "", err
	}

	tokenOptions := func(options *auth.TokenOptions) {
		options.ExpiresIn = expiry
	}

	// Use `auth.GenerateDbConnectAuthToken` if you are _not_ logging in as `admin` user
	token, err := auth.GenerateDBConnectAdminAuthToken(ctx, clusterEndpoint, region, cfg.Credentials, tokenOptions)
	if err != nil {
		return "", err
	}

	return token, nil
}
// --8<-- [end:go-generate-token]

File: python/authentication/src/generate_token.py
# Python SDK examples for generating Aurora DSQL authentication tokens
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# --8<-- [start:python-generate-token]
import boto3

def generate_token(your_cluster_endpoint, region):
    client = boto3.client("dsql", region_name=region)
    # use `generate_db_connect_auth_token` instead if you are not connecting as admin.
    token = client.generate_db_connect_admin_auth_token(your_cluster_endpoint, region)
    print(token)
    return token
# --8<-- [end:python-generate-token]